dropwizard-jdbi/src/main/java/io/dropwizard/jdbi/DBIFactory.java

```diff
@@ -53,7 +57,14 @@ public DBI build(Environment environment,
         final String validationQuery = configuration.getValidationQuery();
         final DBI dbi = new DBI(dataSource);
         environment.lifecycle().manage(dataSource);
-        environment.healthChecks().register(name, new DBIHealthCheck(dbi, validationQuery));
+        ExecutorService executorService = environment.lifecycle().executorService("DBIHealthCheck")
```

Comment:
Shouldn't we consider `ThreadPoolExecutor.CallerRunsPolicy` as a *rejectedExecutionHandler*?
If two health check arrived concurrently, the second one will fail even if the database is up.